### PR TITLE
[#4845] fix(core): Fix bugs when updating metalake or catalog with nullable comment.

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Change the update SQL in the mapper to handle nullable comment. 

### Why are the changes needed?

It's a bug to fix.

Fix: #4845 

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

Added UTs and ITs

### CODE REVIEW

**Comment by jerryshao** [comment]
Do we have similar issue for "topic"?

**Comment by yuqi1129** [comment]
> Do we have similar issue for "topic"?

This problem only occurs in conditions like where or on predicate and the root cause is 

```
select null = null 
``` 
will return false. 


The update method for "topic" is `insert into xxx on duplicate key update set....,comment = xxx` and has no similar problem.

**Comment by yuqi1129** [comment]
This is another PR for #4541.

File: core/src/main/java/org/apache/gravitino/storage/relational/mapper/CatalogMetaMapper.java [hunk 1]
```diff
@@ -150,7 +150,8 @@ CatalogPO selectCatalogMetaByMetalakeIdAndName(
           + " AND metalake_id = #{oldCatalogMeta.metalakeId}"
           + " AND type = #{oldCatalogMeta.type}"
           + " AND provider = #{oldCatalogMeta.provider}"
-          + " AND catalog_comment = #{oldCatalogMeta.catalogComment}"
+          + " AND (catalog_comment = #{oldCatalogMeta.catalogComment} "
+          + "   or (catalog_comment is null and #{oldCatalogMeta.catalogComment} is null))"
```
Comment (jerryshao) [inline core/src/main/java/org/apache/gravitino/storage/relational/mapper/CatalogMetaMapper.java:154]: Can you please change to "IS NULL" for reserved words?

Comment (yuqi1129) [reply core/src/main/java/org/apache/gravitino/storage/relational/mapper/CatalogMetaMapper.java]: DONE

**Review by jerryshao — APPROVED** [review]

**Pull request merged**